test/test_euler.py

```diff
@@ -97,23 +98,42 @@ def test_inviscid_flux(actx_factory, dim):
             for i in range(dim)
         ]
     )
+    massfracs = None
+    if nspecies > 1:
+        massfracs = make_obj_array(
+            [
+                cl.clrandom.rand(
+                    actx.queue, (mesh.nelements,), dtype=np.float64
+                )
+                for i in range(nspecies)
+            ]
+        )
 
-    q = join_conserved(dim, mass=mass, energy=energy, momentum=mom)
+    q = join_conserved(dim, mass=mass, energy=energy, momentum=mom,
+                       massfractions=massfracs)
     cv = split_conserved(dim, q)
 
     # {{{ create the expected result
 
     p = eos.pressure(cv)
     escale = (energy + p) / mass
 
-    expected_flux = np.zeros((dim + 2, dim), dtype=object)
+    numeq = dim + 2
+    if nspecies > 1:
+        numeq = dim + 2 + nspecies
```

Comment:
Redundant logic.

Reply:
Unneeded at least.    I think it's better now?